File: GameShelf/Models/Store.swift
//
//  Store.swift
//  GameShelf
//

import Foundation

/// Tienda de la que proviene un juego.
///
/// Se guarda como `String` en SwiftData (no como entero) para que agregar o
/// reordenar casos no corrompa los datos ya guardados.
enum Store: String, Codable, CaseIterable, Sendable {
  case steam
  case psn
  case epic

  /// Nombre para mostrar en pantalla.
  var displayName: String {
    switch self {
    case .steam: "Steam"
    case .psn: "PlayStation Network"
    case .epic: "Epic Games"
    }
  }
}

File: android/app/src/main/java/com/gameshelf/domain/Store.kt
package com.gameshelf.domain

/**
 * Tienda de la que proviene un juego.
 *
 * Se persiste por su [id] de texto (no por el ordinal del enum) para que
 * agregar o reordenar casos no corrompa los datos ya guardados. Es la misma
 * decision que en la version de iOS.
 */
enum class Store(val id: String) {
  STEAM("steam"),
  PSN("psn"),
  EPIC("epic");

  /**
   * Nombre para mostrar en pantalla.
   *
   * Son nombres propios de marca: no se traducen, por eso viven aca y no en
   * `strings.xml`.
   */
  val displayName: String
    get() = when (this) {
      STEAM -> "Steam"
      PSN -> "PlayStation Network"
      EPIC -> "Epic Games"
    }

  companion object {
    /** Busca una tienda por su id guardado. `null` si no se reconoce. */
    fun fromId(id: String): Store? = entries.firstOrNull { it.id == id }
  }
}
